nearcore/src/runtime/tests.rs

```diff
@@ -1444,3 +1479,236 @@ fn test_genesis_hash() {
+            env.runtime.genesis_config.min_gas_price,
+            env.runtime.genesis_config.gas_limit,
+            &env.head.epoch_id,
+            0,
+            storage_config,
+            env.head.height + 1,
+            &mut PoolIteratorWrapper::new(&mut transaction_pool),
+            &mut |tx: &SignedTransaction| -> bool {
+                chain
+                    .chain_store()
+                    .check_transaction_validity_period(
+                        &chain.get_block_header(&env.head.prev_block_hash).unwrap(),
+                        &tx.transaction.block_hash,
+                        chain.transaction_validity_period,
+                    )
+                    .is_ok()
+            },
+            PROTOCOL_VERSION,
+            default_produce_chunk_add_transactions_time_limit(),
+        )
+        .unwrap();
+
+    assert_eq!(proposed_transactions.transactions.len(), transactions_count);
+    assert!(proposed_transactions.storage_proof.is_some());
+
+    let validator_storage_config = RuntimeStorageConfig {
+        state_root: env.state_roots[0],
+        use_flat_storage: true,
+        source: StorageDataSource::Recorded(PartialStorage {
+            nodes: proposed_transactions.storage_proof.unwrap(),
+        }),
+        state_patch: Default::default(),
+        record_storage: false,
+    };
+
+    let validated_transactions = env
+        .runtime
+        .prepare_transactions(
+            env.runtime.genesis_config.min_gas_price,
+            env.runtime.genesis_config.gas_limit,
+            &env.head.epoch_id,
+            0,
+            validator_storage_config,
+            env.head.height + 1,
+            &mut TransactionGroupIteratorWrapper::new(&proposed_transactions.transactions),
+            &mut |tx: &SignedTransaction| -> bool {
+                chain
+                    .chain_store()
+                    .check_transaction_validity_period(
+                        &chain.get_block_header(&env.head.prev_block_hash).unwrap(),
+                        &tx.transaction.block_hash,
+                        chain.transaction_validity_period,
+                    )
+                    .is_ok()
+            },
+            PROTOCOL_VERSION,
+            default_produce_chunk_add_transactions_time_limit(),
+        )
+        .unwrap();
+
+    assert_eq!(validated_transactions.transactions, proposed_transactions.transactions);
+}
+
+/// Check that transactions validation fails if provided empty storage proof.
+#[test]
+fn test_prepare_transactions_empty_storage_proof() {
+    let (env, chain, mut transaction_pool) = get_test_env_with_chain_and_pool();
+    let transactions_count = transaction_pool.len();
+
+    let storage_config = RuntimeStorageConfig {
+        state_root: env.state_roots[0],
+        use_flat_storage: true,
+        source: StorageDataSource::Db,
+        state_patch: Default::default(),
+        record_storage: true,
+    };
+
+    let proposed_transactions = env
```

Comment:
code to generate `proposed_transactions` seems to be very similar to the previous test, consider extracting that to a shared function to avoid code duplication